src/main/scala/org/broadinstitute/dsde/firecloud/service/StatusService.scala

```diff
@@ -1,43 +1,61 @@
 
-import java.text.SimpleDateFormat
-import java.util.Date
-
 import akka.actor.Actor
-import spray.json._
-
-import spray.routing._
-
-
-import org.broadinstitute.dsde.firecloud.FireCloudConfig
-
-class StatusServiceActor extends Actor with StatusService {
-  def actorRefFactory = context
-  def receive = runRoute(routes)
+import akka.actor.Props
+import akka.pattern._
+import org.broadinstitute.dsde.firecloud.Application
+import org.broadinstitute.dsde.firecloud.dataaccess.{AgoraDAO, RawlsDAO, SearchDAO, ThurloeDAO}
+import org.broadinstitute.dsde.firecloud.model.SystemStatus
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+import org.broadinstitute.dsde.firecloud.service.StatusService.CollectStatusInfo
+import spray.httpx.SprayJsonSupport
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.impStatus
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
+
+/**
+ * Created by anichols on 4/5/17.
+ */
+object StatusService {
+  def props(statusServiceConstructor: () => StatusService): Props = {
+    Props(statusServiceConstructor())
+  }
+
+  def constructor(app: Application)()(implicit executionContext: ExecutionContext): StatusService = {
+    new StatusService(app.searchDAO, app.agoraDAO, app.thurloeDAO, app.rawlsDAO)
+  }
+
+  case class CollectStatusInfo()
 }
 
-trait StatusService extends HttpService with PerRequestCreator with FireCloudDirectives {
-
-  private final val ApiPrefix = "status"
-  private final val dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
-  private implicit val executionContext = actorRefFactory.dispatcher
-
-  val routes: Route =
-    pathPrefix(ApiPrefix) {
-      pathEnd {
-        respondWithJSON {
-          complete {
-            JsObject(
-              "workspacesUrl" -> JsString(FireCloudConfig.Rawls.baseUrl + "/workspaces"),
-              "methodsUrl" -> JsString(MethodsService.remoteMethodsUrl),
-              "timestamp" -> JsString(dateFormat.format(new Date()))
-            ).toString
-          }
-        }
-      } ~ path("ping") {
-        complete {
-          dateFormat.format(new Date())
-        }
-      }
+class StatusService (val searchDAO: SearchDAO,
+                     val agoraDAO: AgoraDAO,
+                     val thurloeDAO: ThurloeDAO,
+                     val rawlsDAO: RawlsDAO)
+                    (implicit protected val executionContext: ExecutionContext) extends Actor with SprayJsonSupport {
+
+  def collectStatusInfo(): Future[PerRequestMessage] = {
+    for {
+      rawlsOK <- rawlsDAO.status
+      (thurloeOK, thurloeMessage) <- thurloeDAO.status
+      (agoraOK, agoraMessage) <- agoraDAO.status
+      searchOK <- Future(searchDAO.indexExists())
+    } yield {
+      val allOK = rawlsOK && thurloeOK && agoraOK && searchOK
+      var messages: ListBuffer[String] = ListBuffer()
```

Comment:
`vars` in scala make me queasy. You're doing this because each of the status methods return something different and you need to do some finagling to get the right data out.

Suggestion: Change the status to return a case class, i.e. `StatusResponseResult` which has not only the boolean, but the response message. Do the same thing for `SearchDAO`. Maybe add a status trait that each of the daos can call that does this.

Then, here in this collect status info message, you can flatmap over a list of all of the status calls. Disclaimer, haven't tested this
```
    val messages: Seq[String] = List(rawlsDAO.status, thurloeDAO.status, agoraDAO.status, searchDAO.status) map { statusCall =>
      statusCall.status match {
        case false => statusCall.message
      }
    }
```